test/nsidc-smap/nsidc-smap_Regression.ipynb

```diff
@@ -0,0 +1,354 @@
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d056598e-5058-470c-83f9-f8dca360308a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from test_configuraton import non_production_configuration, production_configuration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1bcb56c0-f6f6-411a-9bfd-3f867bfe235a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "environment_configuration = {\n",
+    "    \"https://harmony.earthdata.nasa.gov\": {\n",
+    "        **production_configuration,\n",
+    "        \"env\": Environment.PROD,\n",
+    "    },\n",
+    "    \"https://harmony.uat.earthdata.nasa.gov\": {\n",
+    "        **non_production_configuration,\n",
+    "        \"env\": Environment.UAT,\n",
+    "    },\n",
+    "    \"https://harmony.sit.earthdata.nasa.gov\": {\n",
+    "        **non_production_configuration,\n",
+    "        \"env\": Environment.SIT,\n",
+    "    },\n",
+    "    \"http://localhost:3000\": {\n",
+    "        **non_production_configuration,\n",
+    "        \"env\": Environment.LOCAL,\n",
+    "    },\n",
+    "}\n",
+    "\n",
+    "configuration = environment_configuration.get(harmony_host_url)\n",
+    "\n",
+    "if configuration is not None:\n",
+    "    harmony_client = Client(env=configuration[\"env\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b115ba16-0d31-43b0-9f57-6a7061b7eeda",
+   "metadata": {},
+   "source": [
+    "### Run Tests"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e0788c88-6aee-450a-be35-087ba36a1d80",
+   "metadata": {},
+   "source": [
+    "The next cell runs through each of the tests forming requests that are submitted to Harmony and comparing the downloaded results against reference data files that have been verified.  This ensures that Harmony continues to return the expected binary files for expected requests.\n",
+    "\n",
+    "\n",
+    "The tests include:\n",
+    "\n",
+    "- subset_bounding_box: SPL2SMA, SPL3FTP_E\n",
+    "- subset_by_geojson: SPL2SMA\n",
+    "- subset_by_geojson: SPL2SMA\n",
+    "- subset_by_shapefile: SPL3FTP\n",
+    "- subset_by_variable: SPL2SMP_E, SPL2SMP_E_2\n",
+    "- subset_by_kml: SPL3SMP\n",
+    "- reprojection_to_geographic: SPL4CMDL\n",
+    "            \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e745045b-10fc-4e6c-808c-3bdaf2725d03",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if configuration is not None:\n",
+    "    with TemporaryDirectory() as tmp_dir:\n",
+    "\n",
+    "        for test_name, test_configs in configuration[\"single_output_tests\"].items():\n",
+    "            print(f\"Running Tests: {test_name}\")\n",
+    "\n",
+    "            for shortname, test_config in test_configs.items():\n",
+    "                print(f\"running request: {shortname}\")\n",
```

Comment:
I think we had this conversation on a different PR, but I think from a sense of test output isolation it makes more sense to put the creation of the `TemporaryDirectory` inside the for loops. The test files _should_ all be uniquely named, but I get a bit squeamish, and having a distinct `TemporaryDirectory` per test feels just a little safer.

It's not a big deal, and if you have strong opinions against that change, then that's fine, but I guess I'm not so sure the value of having a `TemporaryDirectory` in a one-shot container, if everything is going inside that same directory.